library/ssl_srv.c

```diff
@@ -840,6 +840,7 @@ static int ssl_pick_cert( mbedtls_ssl_context *ssl,
 
     for( cur = list; cur != NULL; cur = cur->next )
     {
+        flags = 0;
```

Comment:
Do we need to initialize this every loop iteration? If flags is not read (only written to), we can avoid undefined behavior by initializing it once outside the loop.

Reply:
This fix is a step further, to fool-proof it for the future. Citing the commit message:
With this fix, after each iteration, the flags will apply only to the most recent certificate, not to any of the previous ones checked.

Reply:
But is it correct? If the leaf certificate has a flag that allows DTLS, but a parent certificate has a flash that disallows DTLS, should we allow DTLS? Is it correct to ignore flags from previous certificates in the chain?

Reply:
I assumed that the loop will be broken once a faulty flag is found, and understood that that's why Hanno suggested to

> Initialize flags in each iteration of the loop

Reply:
OK